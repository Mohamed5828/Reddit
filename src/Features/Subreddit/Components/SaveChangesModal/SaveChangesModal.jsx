import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import {XButton, Title} from "Features/Subreddit/Components/SelectYourFlair/SelectYourFlair.styled";
import {MdOutlineClear} from "react-icons/md"
import { StyledModal } from "./saveChangesModal.styled";

/**
 * modal appear when unsaved changes blured
 *
 * @param {boolean} show - boolean to show modal
 * @param {function} onDiscard - handle on click discard
 * @param {function} onSave - handle on click save
 * @param {function} onHide - handle on click x
 * @param {boolean} showX - boolean to show x button or not
 * @returns {React.Component}
 */
const SaveChangesModal = ({show, onDiscard, onSave,onHide,showX}) => {
  return (
    <StyledModal
      // {...props}
      show={show}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header closeButton={false}>
        {/* <Modal.Title id="contained-modal-title-vcenter">
          Save changes before leaving?
        </Modal.Title> */}
        <Title>Save changes before leaving?</Title>
        {showX && <XButton onClick={onHide}>
          <MdOutlineClear />
        </XButton>}
      </Modal.Header>
      <Modal.Body>
        <p>
          You have made some changes to your community, do you wish to leave
          this menu without saving?
        </p>
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={onDiscard} className="discard">
          Discard
        </Button>
        <Button variant="primary" className="save" onClick={onSave}>
          Save
        </Button>
      </Modal.Footer>
    </StyledModal>
  );
};

export default SaveChangesModal;
