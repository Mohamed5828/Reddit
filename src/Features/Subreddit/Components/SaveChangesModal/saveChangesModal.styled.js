import Modal from "react-bootstrap/Modal";
import styled from "styled-components";

export const StyledModal = styled(Modal)`
  @media (min-width: 992px) {
    .modal-lg,
    .modal-xl {
      width: 538px;
    }
  }
  justify-content: center;
  div.modal-content {
    max-width: 538px;
    min-width: 410px;
  }

  div.modal-header {
    border-bottom: 1px solid ${({ theme }) => theme.lineColor.primary};
    border-top-right-radius: 4px;
    border-top-left-radius: 4px;
    padding: 16px;
  }
  div.modal-title.h4 {
    color: ${({ theme }) => theme.color.primary};
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
  }

  div.modal-footer {
    background-color: ${({ theme }) => theme.lineColor.primary};
  }

  button.btn-primary {
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
    padding: 4px 16px;
    align-items: center;
    border-radius: 20px;
    box-sizing: border-box;
    text-align: center;
    width: auto;
    justify-content: center;
    display: flex;
  }

  button.btn.discard {
    background-color: transparent;
    color: ${({ theme }) => theme.color.secondary};

    :hover {
      border-color: rgba(135, 138, 140, 0.8);
      color: rgba(135, 138, 140, 0.8);
    }
  }

  button.btn.save {
    background-color: ${({ theme }) => theme.color.secondary};

    :hover {
      background-color: #1483d6;
    }
  }

  /*
  color: #1c1c1c;
  font-size: 16px;
font-weight: 500;
line-height: 20px;
contained-modal-title-vcenter
  */
`;
