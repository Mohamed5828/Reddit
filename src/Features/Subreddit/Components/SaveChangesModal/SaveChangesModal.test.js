import Adapter from "@zarconontol/enzyme-adapter-react-18";
import Enzyme, { shallow } from "enzyme";
import SaveChangesModal from "./SaveChangesModal";

/**
 * @param {boolean} show - boolean to show modal
 * @param {function} onDiscard - handle on click discard
 * @param {function} onSave - handle on click save
 * @param {function} onHide - handle on click x
 * @param {boolean} showX - boolean to show x button or not
 */
function out() {
  console.log("clicked");
}

Enzyme.configure({ adapter: new Adapter() });
describe("modal to check unsaved changes", () => {
  it("should render without crashing", () => {
    expect(
      shallow(
        <SaveChangesModal
          show={true}
          onDiscard={out}
          onSave={out}
          onHide={out}
          showX={true}
        />
      )
    ).toMatchSnapshot();
  });
});
