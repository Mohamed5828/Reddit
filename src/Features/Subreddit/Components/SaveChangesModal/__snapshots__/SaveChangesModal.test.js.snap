// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal to check unsaved changes should render without crashing 1`] = `
<Styled(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  show={true}
  size="lg"
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <styled.div>
      Save changes before leaving?
    </styled.div>
    <styled.button
      onClick={[Function]}
    >
      <MdOutlineClear />
    </styled.button>
  </ModalHeader>
  <ModalBody>
    <p>
      You have made some changes to your community, do you wish to leave this menu without saving?
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className="discard"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Discard
    </Button>
    <Button
      active={false}
      className="save"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Save
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
